### PR TITLE
user can make/cancel book request

### PR DESCRIPTION
**Pull request by nguyenngodinh**

### CODE REVIEW

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -24,11 +24,13 @@
...
-          <li>
-            <%= link_to '#' do %>
-              Request
-            <% end %>
-          </li>
+          <% if user_signed_in? %>
+            <li>
+              <%= link_to book_requests_path do %>
+                Request
+              <% end %>
```
Comment (dungdm93) [inline app/views/layouts/_header.html.erb]: HTML close tag error.
<% end %> after </li>

Comment (dungdm93) [inline app/views/layouts/_header.html.erb]: refactor code:
```
<%= link_to "Request", book_requests_path %>
```

File: db/seeds.rb [hunk 2]
```diff
@@ -56,4 +56,13 @@
   following.each {|followed| user.follow followed}
   followers.each {|follower| follower.follow user}
 
+  users = User.order(:created_at).take 6
+  50.times do
+    description = Faker::Lorem.sentence 5
```
Comment (dungdm93) [inline db/seeds.rb]: Refactor code:
```
Faker::Lorem.paragraph
```

File: app/controllers/book_requests_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  before_action :user_signed_in?, only: [:create, :destroy, :index]
+  
+  def create
+    @book_request = current_user.book_requests.build book_request_params
+    if @book_request.save
+      flash[:success] = "Book request created!"
+      redirect_to book_requests_path
+    else
+      @book_requests = current_user.book_requests
+      render 'book_requests/index'
```
Comment (dungdm93) [inline app/controllers/book_requests_controller.rb]: Refector code:
```
render :index
```

File: app/controllers/book_requests_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    if user_signed_in?
+      @book_requests = current_user.book_requests
+      @book_request = current_user.book_requests.build
+    else
+      redirect_to root_url
+    end
+  end
+
+  def destroy
+    BookRequest.find(params[:id]).destroy
```
Comment (dungdm93) [inline app/controllers/book_requests_controller.rb]: You not handle ```BookRequest.find(params[:id]) return nil``` yet. :-1:

File: db/migrate/20150410061227_create_book_requests.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class CreateBookRequests < ActiveRecord::Migration
+  def change
+    create_table :book_requests do |t|
+      t.integer :isbn
+      t.text :description
+      t.boolean :state
```
Comment (dungdm93) [inline db/migrate/20150410061227_create_book_requests.rb]: ```state``` should be ```integer``` for number of state such as:
```
REQUEST | PENDING | WAITING | ACCEPT | REJECT | N/A
```

File: db/migrate/20150410061227_create_book_requests.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      t.references :user, index: true
+
+      t.timestamps null: false
+    end
+    add_foreign_key :book_requests, :users
+    add_index :book_requests, [:user_id, :created_at]
+    change_column :book_requests, :state, :boolean, :default => false
```
Comment (dungdm93) [inline db/migrate/20150410061227_create_book_requests.rb]: Don't need, remove it.
Change it directly in ```create_table```

File: app/controllers/book_requests_controller.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    if user_signed_in?
+      @book_requests = current_user.book_requests
+      @book_request = current_user.book_requests.build
+    else
+      redirect_to root_url
+    end
+  end
+
+  def destroy
+
```
Comment (dieunb) [inline app/controllers/book_requests_controller.rb]: remove this line

File: app/controllers/book_requests_controller.rb [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      flash[:success] = "Book request deleted!"
+    else
+      flash[:danger] = "Could not delete!"
+    end
+    redirect_to book_requests_path
+  end
+
+  private
+
+    def book_request_params
```
Comment (dieunb) [inline app/controllers/book_requests_controller.rb]: indent

File: app/controllers/book_requests_controller.rb [hunk 9]
```diff
@@ -0,0 +1,41 @@
+class BookRequestsController < InheritedResources::Base
+  
+  before_action :user_signed_in?, only: [:create, :destroy, :index]
+  
+  def create
```
Comment (dieunb) [inline app/controllers/book_requests_controller.rb]: show reorder any actions: index show new edit create update destroy

File: app/models/book_request.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+class BookRequest < ActiveRecord::Base
+  
+  belongs_to :user
+  default_scope -> {order created_at: :desc}
```
Comment (dieunb) [inline app/models/book_request.rb]: remove, don't use

File: app/views/shared/_error_messages.html.erb [hunk 11]
```diff
@@ -1,7 +1,11 @@
 <% if object.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
+<<<<<<< HEAD
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: conflict code

**Pull request merged**